components/UI/InviteUsersToMatrixRoom/index.js

```diff
@@ -0,0 +1,160 @@
+  height: 100%;
+  padding: 0 var(--margin);
+
+  @media ${breakpoints.tabletAndAbove} {
+    padding: 0 calc(var(--margin) * 1.5);
+  }
+
+  h3 {
+    line-height: calc(var(--margin) *3);
+  }
+`;
+
+const FeedbackWrapper = styled.div`
+  margin-top: var(--margin);
+`;
+
+export const InviteUserToMatrixRoom = ({ roomId, onSuccess }) => {
+    const auth = useAuth();
+    const matrixClient = auth.getAuthenticationProvider('matrix').getMatrixClient();
+    const [searchResults, setSearchResults] = useState([]);
+    const { t } = useTranslation('invitationModal');
+    const [userFeedback, setUserFeedback] = useState('');
+    const [errorFeedback, setErrorFeedback] = useState([]);
+
+    const handleChange = (searchString) => {
+        debouncedFetchUsersForContributorSearch(searchString);
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const debouncedFetchUsersForContributorSearch = useCallback(debounce((val) => fetchUsersForContributorSearch(val), 300), []);
+
+    const fetchUsersForContributorSearch = useCallback(async (a) => {
+        try {
+            const users = await matrixClient.searchUserDirectory({ term: a });
+            // always filter ourselves; we most likely do not want to invite ourselves to something, i guess?!
+            const usersWithoutMyself = _.filter(users.results, (user) => user.user_id !== matrixClient.getUserId());
+            // we only update the state if the returned array has entries, to be able to check if users a matrix users or not further down in the code (otherwise the array gets set to [] as soon as you selected an option from the datalist)
+            // const filterResults = users.results.filter(item => _.isEqual(item, option));
+            setSearchResults(usersWithoutMyself);
+        } catch (err) {
+            logger.error(t('Error while trying to fetch users: ') + err);
+        }
+    }, [matrixClient, t]);
+
+    function clearInputs() {
+        setUserFeedback('');
+        setSearchResults([]);
+    }
+
+    const handleInvite = async (selectedUsers) => {
+        setErrorFeedback([]);
+        const errors = [];
+
+        for (const user of selectedUsers) {
+            await matrixClient.invite(roomId, user.user_id)
+                .catch(async error => {
+                    // avoid adding duplicates
+                    if (errors.includes(error.data.error)) return;
+                    errors.push(error.data.error);
+                });
+        }
+
+        if (errors.length !== 0) {
+            // if something went wrong we display the errors and clear all inputs
+            setErrorFeedback(errors);
+        }
+
+        const successAmount = selectedUsers.length - errors.length;
+
+        // if everything is okay, we let the user know and exit the view.
+        successAmount > 0 && setUserFeedback(<Trans t={t} i18nKey="invitedUser" count={successAmount}>{ { successAmount } } user was invited and needs to accept your invitation</Trans>);
```

Comment:
A cleaner way of combining translations with feedback messages would be to set the state here, and then somewhere down use the `<Trans>` component (or `t()` function respectively). For example, you could rename the state to be called something like `invitedUsersCount` and then at the bottom, where we render, instead of

```jsx
{ userFeedback && _.isEmpty(errorFeedback) ? <div>{ userFeedback }</div> :
```

... you would ...
```jsx
{ invitedUsersCount && <Trans ... count={invitedUsersCount}>bla bla bla</Trans> }
```

This would also allow to switch languages _while_ the message is showing up.
Right now we're writing a translated string into the state, which is not really what the state is made for.